tests/test_check_runner.py

```diff
@@ -0,0 +1,145 @@
+    else:
+        base_url_line = ''
+    if api_key:
+        api_key_line = 'api-key: ' + api_key + '\n'
+    else:
+        api_key_line = ''
+
+    config_file = tmpdir_factory.mktemp('data').join('config.ini')
+    config_file.write('[Cachet]\n' + base_url_line + api_key_line)
+
+    return config_file.strpath
+
+
+@pytest.mark.parametrize(
+    'base_url,api_key',
+    [
+        (None, None),
+        (None, 'baby'),
+        ('donthurtme', None),
+    ]
+)
+def test_config_params(tmpdir_factory, mocker, base_url, api_key):
+    """Assert error is raised if a required parameter is missing from the config file"""
+
+    print('base_url = %s, api_key = %s', base_url, api_key)
+
+    config_file = config(tmpdir_factory, base_url=base_url, api_key=api_key)
+
+    with pytest.raises(ValueError):
+        unit.main(script=mocker.Mock(), component_id=99, config_file=config_file)
+
+
+@pytest.mark.parametrize(
+    'script_return,expected_update,n_retries',
+    [
+        (Codes.OK.value, Codes.OK, 0),
+        (Codes.CRITICAL.value, Codes.CRITICAL, 0),
+        ((Codes.CRITICAL.value, Codes.CRITICAL.value, Codes.CRITICAL.value), Codes.CRITICAL, 2),
+        ((Codes.CRITICAL.value, Codes.CRITICAL.value, Codes.OK.value), Codes.OK, 2),
+        ((Codes.CRITICAL.value, Codes.WARNING.value, Codes.WARNING.value), Codes.CRITICAL, 2),
+        ((Codes.CRITICAL.value, Codes.WARNING.value, Codes.OK.value), Codes.OK, 2),
+    ]
+)
+def test_updating(mocker, config, script_return, expected_update, n_retries):
+    """Assert runner updates Cachet with expected codes"""
+
+    try:
+        completed_processes = []
+        for value in script_return:
+            completed_processes.append(CompletedProcessMock(value))
+        mocker.patch('subprocess.run', side_effect=completed_processes)
+    except TypeError:
+        mocker.patch('subprocess.run', return_value=CompletedProcessMock(script_return))
+
+    mocker.patch('cachetclient.cachet.Components.put')
+    component = 99
+
+    with pytest.raises(SystemExit):
+        unit.main(script=mocker.Mock(), component_id=component, config_file=config, script_args=[],
+                  retries=n_retries)
+
+    cachetclient.cachet.Components.put.assert_called_with(
+        id=component,
+        status=unit.codes_mapping[expected_update])
+
+
+def test_incompatible_plugin(mocker, config):
+    """Assert runner fails if monitoring script is not Nagios-compatible"""
+
+    return_code = max(c.value for c in Codes) + 1
+
+    mock_execution = mocker.Mock()
+    mock_execution.returncode = return_code
+
+    mocker.patch('subprocess.run', return_value=mock_execution)
+    mocker.patch('cachetclient.cachet.Components.put')
+
+    common.run_and_assert(unit.main, expected_code=Codes.CRITICAL, script=mocker.Mock(),
+                          component_id=99, config_file=config, script_args=[], retries=0)
+    assert not cachetclient.cachet.Components.put.called
```

Comment:
I'd rather for us to store the result of `mocker.patch` in a variable and then assert on that variable, for better readability. This would have to be done everywhere 😈

Reply:
Done. No assertions are made on other patched methods a part from this one so the fix only applies to this file.